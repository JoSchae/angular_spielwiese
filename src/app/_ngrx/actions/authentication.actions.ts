import { createAction, props } from '@ngrx/store';
import { IAuthentication } from 'src/app/_models/authentication.interface';
import { IError } from 'src/app/_models/error.interface';
import { IAuthenticationState } from '../state/authentication.state';

export const getAuthentication = createAction('[Authentication] Get Authentication');

export const getAuthenticationSuccess = createAction(
    '[Authentication] Get Authentication Success',
    props<{ payload: IAuthenticationState }>()
);

export const getAuthenticationFailure = createAction(
    '[Authentication] Get Authentication Failure',
    props<{ payload: IError }>()
);

export const logout = createAction(
    '[Authentication] Remove authentication state'
);
