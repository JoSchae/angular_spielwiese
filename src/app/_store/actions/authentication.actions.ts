import { Action } from '@ngrx/store';
import { IAuthentication } from 'src/app/_models/authentication.interface';

export enum EAuthenticationActions {
    GetAuthentication = '[Authentication] Get Authentication',
    GetAuthenticationSuccess = '[Authentication] Get Authentication Success',
    GetIsLoggedIn = '[Authentication] Get isLoggedIn',
    GetIsLoggedInSuccess = '[Authentication] Get isLoggedIn Success'
}

export class GetAuthentication implements Action {
    public readonly type = EAuthenticationActions.GetAuthentication;
}

export class GetAuthenticationSuccess implements Action {
    public readonly type = EAuthenticationActions.GetAuthenticationSuccess;
    constructor(public payload: IAuthentication) { }
}

export class GetIsLoggedIn implements Action {
    public readonly type = EAuthenticationActions.GetIsLoggedIn;
    constructor(public payload: string) { }
}

export class GetIsLoggedInSuccess implements Action {
    public readonly type = EAuthenticationActions.GetIsLoggedInSuccess;
    constructor(public payload: boolean) { }
}

export type AuthenticationActions = GetAuthentication | GetAuthenticationSuccess | GetIsLoggedIn | GetIsLoggedInSuccess;
