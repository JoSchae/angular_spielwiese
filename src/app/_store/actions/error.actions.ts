import { Action } from '@ngrx/store';

export enum EErrorAction {
    AddError = '[Error] Add Error'
}

export class AddError implements Action {
    public readonly type = EErrorAction.AddError;
    constructor(public payload: any) { }
}

export type ErrorActions = AddError;
