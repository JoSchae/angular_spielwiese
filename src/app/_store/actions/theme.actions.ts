import { Action } from '@ngrx/store';
import { ITheme } from 'src/app/_models/theme.interface';

export enum EThemeActions {
    GetTheme = '[Theme] Get Theme',
    GetThemeSuccess = '[Theme] Get Theme Success',
    GetThemes = '[Theme] Get Themes',
    GetThemesSuccess = '[Theme] Get Themes Success'
}

export class GetTheme implements Action {
    public readonly type = EThemeActions.GetTheme;
    constructor(public payload: string) { }
}

export class GetThemeSuccess implements Action {
    public readonly type = EThemeActions.GetThemeSuccess;
    constructor(public payload: ITheme) { }
}

export class GetThemes implements Action {
    public readonly type = EThemeActions.GetThemes;
}

export class GetThemesSuccess implements Action {
    public readonly type = EThemeActions.GetThemesSuccess;
    constructor(public payload: ITheme[]) { }
}

export type ThemeActions = GetTheme | GetThemeSuccess | GetThemes | GetThemesSuccess;
