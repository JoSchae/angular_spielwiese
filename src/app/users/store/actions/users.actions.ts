import { createAction, props } from '@ngrx/store';
import { IUser } from 'src/app/_models/user.interface';

export const getUsers = createAction('[Users] Get Users');

export const getUsersSuccess = createAction(
    '[Users] Get Users Success',
    props<{ users: IUser[] }>()
);
