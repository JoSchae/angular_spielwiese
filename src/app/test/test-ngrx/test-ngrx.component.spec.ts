import { TestNgrxComponent } from './test-ngrx.component';
import { createComponentFactory, Spectator, mockProvider } from '@ngneat/spectator/jest';
import { TestDataFacade } from '../stores/ngrx/facades/testdata.facade';
import { TestDataFacadeMock } from '../stores/ngrx/facades/testdata-mock.facade';
import { TestDataFacadeImpl } from '../stores/ngrx/facades/testdata-impl.facade';
import { GETAllTestData } from '../stores/ngrx/actions/testdata.actions';
import { of } from 'rxjs';

describe('TestNgrxComponent', () => {

    const createComponent = createComponentFactory({
        component: TestNgrxComponent,
        componentProviders: [
            {
                provide: TestDataFacade,
                useClass: TestDataFacadeMock
            }
        ]
    });

    let spectator: Spectator<TestNgrxComponent>;
    let component: TestNgrxComponent;
    let service: TestDataFacade;

    beforeEach(() => {
        spectator = createComponent();
        component = spectator.component;
        service = spectator.get(TestDataFacade, true);
        spectator.detectChanges();
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should have all 3 selectors defined', () => {
        expect(component.testData$).toBeDefined();
        expect(component.specificData$).toBeDefined();
        expect(component.testDataById$).toBeDefined();
    });

    it('should match the markup after ngInit', () => {
        expect(spectator.fixture).toMatchSnapshot();
    });
});
