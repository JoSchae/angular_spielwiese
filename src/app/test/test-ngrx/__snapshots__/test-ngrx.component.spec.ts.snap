// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestNgrxComponent should match the markup after ngInit 1`] = `
<tmp-test-ngrx
  _testDataFacade={[Function TestDataFacadeMock]}
  specificData$={[Function Observable]}
  testData$={[Function Observable]}
  testDataById$={[Function Observable]}
>
  <div>
    <p>
      Wir haben 1 Test Daten
    </p>
  </div><div>
    <p>
      Specific Date:
    </p>
    <ul>
      <li>
        johannes
      </li>
      <li>
        schäfer
      </li>
    </ul>
  </div><div>
    <p>
      Data by ID
    </p>
    <ul>
      <li>
        johannes
      </li>
      <li>
        schäfer
      </li>
    </ul>
  </div>
</tmp-test-ngrx>
`;
